dttools/src/Makefile

```diff
@@ -139,6 +140,9 @@ auth_globus.o: auth_globus.c
 auth_test: auth_test.o libdttools.a
 	$(CCTOOLS_LD) -o $@ $(CCTOOLS_INTERNAL_LDFLAGS) $(LOCAL_LDFLAGS) $^ $(LOCAL_LINKAGE) $(CCTOOLS_GLOBUS_LDFLAGS) $(CCTOOLS_EXTERNAL_LINKAGE)
 
+lib_io_helper.so: lib_io_helper.o
+	$(CCTOOLS_CC) -shared -ldl -fPIC $^ -o $@
```

Comment:
For some reason, ubuntu does not like -ldl there. You want:

$(CCTOOLS_CC) -shared -ldl -fPIC $^ -o $@ -ldl